internal/oauth/oauth.go

```diff
@@ -58,10 +58,32 @@ const (
 	GovClientID = "0oabtyfelbTBdoucy297" // GovClientID for production
 )
 
-func changeHostNameToContainer() {
-	if !strings.Contains(config.UserAgent, config.ContainerHostName) {
-		config.UserAgent = strings.ReplaceAll(config.UserAgent, config.HostName, config.ContainerHostName)
-		config.HostName = config.ContainerHostName
+// Patches the agent hostname based on set env vars.
+func patchConfigHostnameFromEnvs(checker envs.EnvChecker) {
+	var builder strings.Builder
+	if checker.IsPopulated(config.AtlasActionHostNameEnv) {
+		appendToHostName(&builder, config.AtlasActionHostName)
+	}
+	if checker.IsPopulated(config.GitHubActionsHostNameEnv) {
+		appendToHostName(&builder, config.GitHubActionsHostName)
+	}
+	if checker.IsPopulated(config.ContainerizedHostNameEnv) {
+		appendToHostName(&builder, config.DockerContainerHostName)
+	}
+	hostnameEnvVal := builder.String()
+
+	if hostnameEnvVal != "" && !strings.Contains(config.UserAgent, hostnameEnvVal) {
+		config.UserAgent = strings.ReplaceAll(config.UserAgent, config.HostName, hostnameEnvVal)
+		config.HostName = hostnameEnvVal
+	}
+}
+
+func appendToHostName(builder *strings.Builder, configVal string) {
+	if configVal != "" {
+		if builder.Len() > 0 {
+			builder.WriteString("|")
+		}
+		builder.WriteString(configVal)
```

Comment:
[nit] removing a layer
```suggestion
	if configVal == "" {
	   return
	}
	if builder.Len() > 0 {
		builder.WriteString("|")
	}
	
	builder.WriteString(configVal)
```

Reply:
Updated, thanks for the suggestion!